`skaffold build` panics when Dockerfile has empty `FROM` instruction

<!--
  Issues without logs and details are more complicated to fix.
  Please help us by filling the template below!
-->

### Expected behavior
Skaffold surfaces a helpful error that tells me what is broken in my Dockerfile

### Actual behavior
Skaffold panics:

```
➜  microservices git:(master) skaffold build
Generating tags...
 - leeroy-web -> gcr.io/nkubala-demo/leeroy-web:v1.15.0-22-ga549c7a1d
 - leeroy-app -> gcr.io/nkubala-demo/leeroy-app:v1.15.0-22-ga549c7a1d-dirty
Checking cache...
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x4bbea4a]

goroutine 39 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.fromInstruction(0xc000926850, 0xc0009e7660, 0x5a57961, 0x7, 0xc0009e7718)
        /Users/nkubala/Development/skaffold/pkg/skaffold/docker/parse.go:361 +0x2a
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.expandOnbuildInstructions(0xc00093c940, 0x8, 0x8, 0x5d99020, 0xc000540840, 0x0, 0x0, 0xc0009e7878, 0x40b069b, 0xc0009300d0)
        /Users/nkubala/Development/skaffold/pkg/skaffold/docker/parse.go:309 +0x14b
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.readCopyCmdsFromDockerfile(0x0, 0xc00016e0c0, 0x5c, 0xc0002a9230, 0xa, 0x0, 0x5d99020, 0xc000540840, 0x0, 0x0, ...)
        /Users/nkubala/Development/skaffold/pkg/skaffold/docker/parse.go:84 +0x345
github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker.GetDependencies(0x5d98b60, 0xc00093c840, 0xc0002a9230, 0xa, 0x5a5b15a, 0xa, 0x0, 0x5d99020, 0xc000540840, 0x400f830, ...)
        /Users/nkubala/Development/skaffold/pkg/skaffold/docker/dependencies.go:61 +0x26b
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build.DependenciesForArtifact(0x5d98b60, 0xc00093c840, 0xc0009265b0, 0x5d99020, 0xc000540840, 0x5448239, 0xc0009e7bf0, 0x404eb85, 0xc00014a7b0, 0x0)
        /Users/nkubala/Development/skaffold/pkg/skaffold/build/dependencies.go:42 +0x4af
github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner.NewForConfig.func1(0x5d98b60, 0xc00093c840, 0xc0009265b0, 0x0, 0x1, 0xc00014a7b0, 0x0, 0x1)
        /Users/nkubala/Development/skaffold/pkg/skaffold/runner/new.go:78 +0x7b
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache.getHashForArtifact(0x5d98b60, 0xc00093c840, 0xc000b39ce0, 0xc0009265b0, 0x5a53fcf, 0x5, 0x0, 0x0, 0x0, 0x0)
        /Users/nkubala/Development/skaffold/pkg/skaffold/build/cache/hash.go:56 +0x20f
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache.NewCache.func1(0x5d98b60, 0xc00093c840, 0xc0009265b0, 0x0, 0x0, 0x0, 0x0)
        /Users/nkubala/Development/skaffold/pkg/skaffold/build/cache/cache.go:100 +0x7c
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache.(*cache).lookup(0xc000950ea0, 0x5d98b60, 0xc00093c840, 0xc0009265b0, 0xc0002da8c0, 0x3a, 0x0, 0x0)
        /Users/nkubala/Development/skaffold/pkg/skaffold/build/cache/lookup.go:51 +0x66
github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache.(*cache).lookupArtifacts.func1(0xc000950ea0, 0x5d98b60, 0xc00093c840, 0xc0003a7520, 0x2, 0x2, 0x1, 0xc0009fe4b0, 0xc000798440, 0x2, ...)
        /Users/nkubala/Development/skaffold/pkg/skaffold/build/cache/lookup.go:41 +0xb5
created by github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/cache.(*cache).lookupArtifacts
        /Users/nkubala/Development/skaffold/pkg/skaffold/build/cache/lookup.go:40 +0x134
```

### Information

- Skaffold version: v1.15.0
- Operating system: OS X
- Contents of skaffold.yaml: Any of our examples

### Steps to reproduce the behavior

1. pick an example with a dockerfile. remove the `FROM` line and replace it with just `FROM` (this is broken I know)
2. `skaffold build`

--- a/pkg/skaffold/docker/dependencies_test.go
+++ b/pkg/skaffold/docker/dependencies_test.go
@@ -216,6 +216,11 @@ FROM stage
 ADD ./file /etc/file
 `
 
+const invalidFrom = `
+FROM
+COPY . /
+`
+
 type fakeImageFetcher struct{}
 
 func (f *fakeImageFetcher) fetch(image string, _ Config) (*v1.ConfigFile, error) {
@@ -529,6 +534,12 @@ func TestGetDependencies(t *testing.T) {
 			ignore:         "bar\ndocker/*",
 			ignoreFilename: "Dockerfile.dockerignore",
 			expected:       []string{".dot", "Dockerfile", "Dockerfile.dockerignore", "file", "server.go", "test.conf", "worker.go"},
+		},
+		{
+			description: "invalid dockerfile",
+			dockerfile:  invalidFrom,
+			workspace:   ".",
+			shouldErr:   true,
 		},
 	}
 

--- a/pkg/skaffold/docker/parse.go
+++ b/pkg/skaffold/docker/parse.go
@@ -27,6 +27,7 @@ import (
 
 	v1 "github.com/google/go-containerregistry/pkg/v1"
 	"github.com/moby/buildkit/frontend/dockerfile/command"
+	"github.com/moby/buildkit/frontend/dockerfile/instructions"
 	"github.com/moby/buildkit/frontend/dockerfile/parser"
 	"github.com/moby/buildkit/frontend/dockerfile/shell"
 	"github.com/sirupsen/logrus"
@@ -72,6 +73,11 @@ func readCopyCmdsFromDockerfile(onlyLastImage bool, absDockerfilePath, workspace
 
 	res, err := parser.Parse(f)
 	if err != nil {
+		return nil, fmt.Errorf("parsing dockerfile %q: %w", absDockerfilePath, err)
+	}
+
+	// instructions.Parse will check for malformed Dockerfile
+	if _, _, err := instructions.Parse(res.AST); err != nil {
 		return nil, fmt.Errorf("parsing dockerfile %q: %w", absDockerfilePath, err)
 	}
 

